src/components/apartaments/Apartments.jsx

```diff
@@ -1,13 +1,162 @@
-import React from 'react';
-
 import './Apartments.css';
+import React, { useEffect, useState } from 'react'
+
+import CardHorizontal from '../cards/CardHorizontal';
+
+import '../cards/Cards.css';
+import '../cards/CardsHorizontal.css';
+import './LoaderIco.css'
+
+import hotelsService from '../../services/hotelsService'
+
+function ApartmentsExp(props) {
+  const [catchProps, setCatchProps] = useState({
+    name: undefined,
+    start: undefined,
+    end: undefined,
+    adult: undefined,
+    child: undefined
+  });
+  const [isSearchSuccesful, setIsSearchSuccesful] = useState(true);
+  const [hotelsList, setHotelsList] = useState([]);
+  const [hotel, setHotel] = useState();
+  const [hotelsBaseInfo, setHotelsBaseInfo] = useState([]);
+  const [counter, setCounter] = useState(-1);
+  useEffect(() => {
+    setHotelsList([])
+    setCounter(-1)
+    if (props.valueFromParent !== undefined) {
+      setCatchProps(props.valueFromParent.data_from_child)
+    } else {
+      setCatchProps(props)
+    }
+  }, [props])
+  console.log(catchProps)
+  useEffect(() => {
+    console.log(catchProps.name)
+    hotelsService.searchHotelByLocation({ query: catchProps.name })
+      .then(({ data }) => {
+        setHotelsBaseInfo(data.data.suggestions[1].entities);
+        setCounter(counter + 1);
+      })
+      .catch((error) => {
+        console.error(error);
+        Promise.resolve({});
+      })
+  }, [catchProps])
+
+  useEffect(() => {
+    if (hotelsBaseInfo[counter] !== undefined) {
+      setIsSearchSuccesful(true);
+      hotelsService.getHotelDetails({
+        id: hotelsBaseInfo[counter].destinationId,
+        checkIn: catchProps.start,
+        checkOut: catchProps.end,
+        adults1: catchProps.adult,
+        child1: catchProps.child
+      })
+        .then(({ data }) => {
+          const newHotel = {
+            name: hotelsBaseInfo[counter].name,
+            overview: data.data.data.body.overview.overviewSections[0].content.join(),
+            address: data.data.data.body.propertyDescription.address.fullAddress,
+            starRating: data.data.data.body.propertyDescription.starRating,
+            guestRating: data.data.data.body.guestReviews.formattedRating
+          };
+          setHotel(newHotel);
+        })
+        .catch((error) => {
+          console.error(error);
+          setIsSearchSuccesful(false);
```

Comment:
error, setError ???

Reply:
this setState is used to check ....    to show the user that the search was successful or uncompleted ... cause if the user enters incorrect data, the api will return an error

Reply:
So you can setError and display "Something went wrong" for example
it is up to you